hasher-matcher-actioner/hmalib/lambdas/integrations.py

```diff
@@ -0,0 +1,200 @@
+import json
+import hmac
+import typing as t
+from dataclasses import dataclass
+
+from hmalib.common.logging import get_logger
+
+import requests
+
+logger = get_logger()
+
+# Placeholder values not used in prod so that we dont leak anything to github
+API_TOKEN = "3156067"
+WORKPLACE_APP_SECRET = "70856018561"
+HMA_API_GATEWAY_URL = "https://xxxxxxxxx.execute-api.us-east-1.amazonaws.com"
+
+
+@dataclass
+class PhotoToUpload:
+    photo_id: str
+    photo_url: str
+    additional_fields: t.List[str]
+
+
+def lambda_handler(event, context):
+    logger.info(event)
+
+    if not can_process_request(event):
+        logger.info("Cannot Process Request")
+        return {"statusCode": 400, "body": json.dumps(event)}
+    if not authorize_request(event):
+        logger.info("Cannot Authorize Request")
+        return {"statusCode": 400, "body": json.dumps("Cannot Authorize Request")}
+    response, content_to_upload = process_request(event)
+    for content in content_to_upload:
+        print(f"Uploading content with ID {content.photo_id} to HMA")
+        upload_to_HMA(content)
+
+    return response
+
+
+def upload_to_HMA(content: PhotoToUpload):
+    submit_payload = {
+        "content_id": content.photo_id,
+        "content_type": "photo",
+        "content_url": content.photo_url,
+        "additional_fields": content.additional_fields,
+        "force_resubmit": True,  # For testing, allow duplicate submission
+    }
+
+    payload_bytes = json.dumps(submit_payload).encode()
+
+    response = requests.post(
+        url=HMA_API_GATEWAY_URL + "/submit/url/",
+        headers={
+            "Content-Type": "application/json",
+            "Authorization": API_TOKEN,
+        },
+        data=payload_bytes,
+    )
+    response.raise_for_status()
+
+    logger.info(response.json())
```

Comment:
This is potentially the only code that would be a helpful example for a partner. Potentially just the `upload_to_HMA` function

Reply:
That does sound right. Unless partners are using lambdas to power their APIs, there is limited value-add for these pieces of code. 

`upload_to_HMA` is something we could provide. Even that is limited to being used by those who use python, but is a reference impl. @BarrettOlson already wrote one when he was writing an e2e test. https://github.com/facebook/ThreatExchange/blob/main/hasher-matcher-actioner/scripts/hma_script_utils.py#L110-L131

That should IMO suffice as a reference impl. @BarrettOlson correct me if I am wrong, but you've also ported the same library to use access tokens instead of the cognito mechanism.

What we _could_ do, to make this more usable, is to provide it in form of a python package. Something like `hmaclient??`. 

Partners could do:
```python
from hmaclient import HMAClient

..
  client = HMAClient(endpoint="https://***.execute.aws", access_token="SDFSF......S"
  client.submit_url(...)
..
```

This way, your integration process might have been smoother. You'd not have to write the upload_to_HMA() function from scratch.

Reply:
I like this. I wonder if its worth the effort however